lib/spack/spack/test/conftest.py

```diff
@@ -190,23 +205,29 @@ def working_env():
 
 @pytest.fixture(scope='function', autouse=True)
 def check_for_leftover_stage_files(request, mock_stage, ignore_stage_files):
-    """Ensure that each test leaves a clean stage when done.
+    """
+    Ensure that each (mock_stage) test leaves a clean stage when done.
 
-    This can be disabled for tests that are expected to dirty the stage
-    by adding::
+    Tests that are expected to dirty the stage can disable the check by
+    adding::
 
         @pytest.mark.disable_clean_stage_check
 
-    to tests that need it.
+    and the associated stage files will be removed.
     """
     stage_path = mock_stage
 
     yield
 
     files_in_stage = set()
-    if os.path.exists(stage_path):
+    try:
         stage_files = os.listdir(stage_path)
         files_in_stage = set(stage_files) - ignore_stage_files
+    except OSError as err:
+        if err.errno == 2:
```

Comment:
Could you replace `2` with the `errno` module variable e.g. `if err.errno == errno.ENOEXEC`?

Reply:
Sure, I can use `errno.ENOENT` instead.